File: VB/MyDateEdit/MyDateEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Reflection
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.Utils.Win
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Popup
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository

Namespace WindowsFormsApplication2

	Public Class MyDateEdit
		Inherits DateEdit
		Shared Sub New()
			RepositoryItemMyDateEdit.Register()
		End Sub
		Public Sub New()
		End Sub

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemMyDateEdit.EditorName
			End Get
		End Property
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemMyDateEdit
			Get
				Return TryCast(MyBase.Properties, RepositoryItemMyDateEdit)
			End Get
		End Property
		Protected Overrides Function CreatePopupForm() As PopupBaseForm
			If Properties.VistaDisplayMode <> DevExpress.Utils.DefaultBoolean.False AndAlso Properties.ShowYearArrows Then
				Return New MyVistaPopupDateEditForm(Me)
			End If
			Return MyBase.CreatePopupForm()
		End Function
	End Class
End Namespace

File: CS/MyDateEdit/MyDateEdit.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.Utils.Win;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Popup;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;

namespace WindowsFormsApplication2 {

    public class MyDateEdit : DateEdit {
        static MyDateEdit() {
            RepositoryItemMyDateEdit.Register();
        }
        public MyDateEdit() { }

        public override string EditorTypeName {
            get { return RepositoryItemMyDateEdit.EditorName; }
        }
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemMyDateEdit Properties {
            get { return base.Properties as RepositoryItemMyDateEdit; }
        }
        protected override PopupBaseForm CreatePopupForm() {
            if (Properties.VistaDisplayMode != DevExpress.Utils.DefaultBoolean.False && Properties.ShowYearArrows) return new MyVistaPopupDateEditForm(this);
            return base.CreatePopupForm();
        }
    }
}
